src/web/services/link.service.ts

```diff
@@ -0,0 +1,139 @@
+import { Injectable } from '@angular/core';
+
+import { Instructor, Student } from '../types/api-output';
+import { environment } from './../environments/environment';
+import { NavigationService } from './navigation.service';
+
+/**
+ * Handles the logic for generating links on the client.
+ */
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LinkService {
+
+  URI_PREFIX: string = 'web';
+  JOIN_PAGE: string = 'join';
+  STUDENT_HOME_PAGE: string = 'student/home';
+  INSTRUCTOR_HOME_PAGE: string = 'instuctor/home';
+  ADMIN_ACCOUNTS_PAGE: string = 'admin/accounts';
+  INSTRUCTOR_STUDENT_RECORDS_PAGE: string = 'instructor/students/records';
+  SESSIONS_SUBMISSION_PAGE: string = 'sessions/submission';
+  SESSIONS_RESULT_PAGE: string = 'sessions/result';
```

Comment:
The URIs here can be added to `java/teammates/ui/webapi/endpoints/ResourceEndpoints` in the backend, and imported from `api-endpoints.ts` after you run `generateTypes`. This will ensure consistency between frontend and backend.